pkgs/crypto/keys/client/common.go

```diff
@@ -21,13 +21,18 @@ var DefaultBaseOptions = BaseOptions{
 
 func HomeDir() string {
 	// if environment set, always use that.
+	// if not, check whether can get os.UserHomeDir()
+	// if not, fall back to home directory
+	var err error
 	hd := os.Getenv("GNO_HOME")
 	if hd != "" {
 		return hd
 	}
-
-	// look for dir in home directory.
-	hd, err := os.UserHomeDir()
+	hd, err = os.UserConfigDir()
```

Comment:
I was looking through the go source, and it looks like the behaviour on non-windows is always a `Getenv("HOME")` for both UserConfigDir and UserHomeDir. Which means that realistically the result of the function is either `hd + "/gno"` or a panic. I think we can change this so that

```
hd, err = ...
if err != nil // panic
return filepath.Join(hd, "gno")
```

also, change variable name to `cd`? (`h` doesn't make sense anymore)

Reply:
they don't use xdg? why did they make an API in that case :roll_eyes: 
ok I will look that up. Thanks for the notice!

Reply:
I took a look the doc https://docs.studygolang.com/pkg/os/#UserConfigDir
but it seems ok to me:

> func UserConfigDir() ([string](https://docs.studygolang.com/pkg/builtin/#string), [error](https://docs.studygolang.com/pkg/builtin/#error))
> 
> UserConfigDir returns the default root directory to use for user-specific configuration data. Users should create their own application-specific subdirectory within this one and use that.
> 
> On Unix systems, it returns $XDG_CONFIG_HOME as specified by https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html if non-empty, else $HOME/.config. On Darwin, it returns $HOME/Library/Application Support. On Windows, it returns %AppData%. On Plan 9, it returns $home/lib.
> 
> If the location cannot be determined (for example, $HOME is not defined), then it will return an error.

And 

> 
> 
> func [UserHomeDir](https://docs.studygolang.com/src/os/file.go?s=14763:14797#L490) 1.12
> 
> func UserHomeDir() ([string](https://docs.studygolang.com/pkg/builtin/#string), [error](https://docs.studygolang.com/pkg/builtin/#error))
> 
> UserHomeDir returns the current user's home directory.
> 
> On Unix, including macOS, it returns the $HOME environment variable. On Windows, it returns %USERPROFILE%. On Plan 9, it returns the $home environment variable.